flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -237,66 +253,11 @@ void testDelayedGrantCallAfterContenderBeingDeregisteredAgain() throws Exception
         };
     }
 
-    /**
-     * Test to cover the issue described in FLINK-31814. This test could be removed after
-     * FLINK-31814 is resolved.
```

Comment:
We aren't resolving FLINK-31814 in this ticket; why are we removing this test?

Reply:
I noticed that the behavior was actually already fixed by (re-)introducing the `DefaultLeaderElectionService.running` property. Close will set the property to false which would disable the revoke event processing in [DefaultLeaderElectionService:468](https://github.com/apache/flink/blob/4338f36dee313ca5d5901a753961323e9f299372/flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionService.java#L468). That was why I removed the test. The kept the Jira issue FLINK-31814 to deal with the precondition